pkg/configs/advisory/v2/detection.go

```diff
@@ -124,3 +136,51 @@ func (d DetectionNVDAPI) Validate() error {
 		),
 	)
 }
+
+var (
+	DetectionScannerGrype = "grype"
+)
+
+var DetectionScanners = []string{
+	DetectionScannerGrype,
+}
+
+// DetectionScanV1 is the data associated with DetectionTypeScanV1.
+type DetectionScanV1 struct {
+	SubpackageName    string `yaml:"subpackageName"`
+	ComponentID       string `yaml:"componentID"` // TODO: consider namespacing this ID using the SBOM tool+format
+	ComponentName     string `yaml:"componentName"`
+	ComponentVersion  string `yaml:"componentVersion"`
+	ComponentType     string `yaml:"componentType"`
+	ComponentLocation string `yaml:"componentLocation"`
+	Scanner           string `yaml:"scanner"` // TODO: it'd be nice for the scanner value to be automatically versioned
+}
+
+// Validate returns an error if the DetectionScanV1 data is invalid.
+func (d DetectionScanV1) Validate() error {
+	// TODO: Should SubpackageName be required, and it's set to the origin package
+	// 	sometimes? Or should an empty value imply this was a scan of an origin
+	// 	package?
```

Comment:
Surfacing this question for PR discussion!

Reply:
I'd require it for consistency but I don't have a really strong opinion!

Reply:
yeah let's always set it

Reply:
Done!